lib/ansible/modules/cloud/azure/azure_rm_securitygroup_facts.py

```diff
@@ -60,21 +60,47 @@
 
 RETURN = '''
 azure_securitygroups:
-    description: List containing security group dicts.
+    description:
+        - List containing security group dicts.
     returned: always
-    type: list
-    example: [{
-        "etag": 'W/"d036f4d7-d977-429a-a8c6-879bc2523399"',
-        "id": "/subscriptions/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx/resourceGroup/myResourceGroup/providers/Microsoft.Network/networkSecurityGroups/secgroup001",
-        "location": "eastus2",
-        "name": "secgroup001",
-        "properties": {
-            "defaultSecurityRules": [
-                {
-                    "etag": 'W/"d036f4d7-d977-429a-a8c6-879bc2523399"',
-                    "id": "/subscriptions/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx/resourceGroup/myResourceGroup/providers/Microsoft.Network/networkSecurityGroups/secgroup001/defaultSecurityRules/AllowVnetInBound",
-                    "name": "AllowVnetInBound",
-                    "properties": {
+    type: complex
```

Comment:
type list

Reply:
if describe nested return values, set type: complex and repeat the elements above for each sub-field.